### PR TITLE
HOLA-106 bulk minter extra file support

### PR DESCRIPTION
**Pull request by davetorbeck**
https://holaplex.atlassian.net/browse/HOLA-106?atlOrigin=eyJpIjoiOTBmYmViMzdmZDk5NDYyYzhmZTRhMGU3OGZjNGEwN2EiLCJwIjoiaiJ9

#### TODO:
fill out changes

### CODE REVIEW

**Comment by davetorbeck** [comment]
Anyone know why I got such a massive diff when I merged in `dev` with the `yarn.lock`?

File: pages/nfts/new.tsx [hunk 1]
```diff
@@ -65,79 +72,112 @@ export interface Collection {
...
           }
         : state;
+    }
+    case 'SET_FILE_PREVIEWS':
+      return { ...state, filePreviews: action.payload as FilePreview[] };
+    case 'INSERT_COVER_IMAGE': {
+      const { coverImage, index } = action.payload as { coverImage: File; index: number };
+
+      const copy = { ...state.filePreviews[index], coverImage };
+      console.log({ copy });
```
Comment (kespinola) [inline pages/nfts/new.tsx:172]: drop console log.

File: pages/nfts/new.tsx [hunk 2]
```diff
@@ -166,42 +206,93 @@ export default function BulkUploadWizard() {
...
+        if (!filePin) {
+          throw new Error('No file pin for index ' + i);
+        }
+
+        const category: NFTCategory = detectCategoryByFileExt(filePin.name);
+        const isMultiMedia = category !== 'image';
+
+        let coverImageFile: UploadedFilePin | undefined;
+        if (v.coverImageFile) {
+          coverImageFile = await uploadCoverImage(v.coverImageFile);
```
Comment (kespinola) [inline pages/nfts/new.tsx:242]: Moving upload out of this transform would separate the concerns. Ideally transform can happen synchronously.

File: src/common/components/elements/ModelViewer.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import '@google/model-viewer';
+
+const ModelViewer = (props: any) => (
+  // @ts-ignore
+  <model-viewer {...props} style={{ width: 800, height: 400 }}></model-viewer>
```
Comment (kespinola) [inline src/common/components/elements/ModelViewer.tsx:6]: This is going to cause mobile viewing issues. Ideal is having model view take the width of the container and the height is the same as the width (square). Use padding trick leveraged by image previews to achieve.

File: src/common/components/elements/NFTPreviewGrid.tsx [hunk 4]
```diff
@@ -76,49 +74,162 @@ const getOverlayStatus = (index: number, nftValues?: NFTValue[]) => {
...
+    showModal();
+  };
+
+  const getFilePreview = (fp: FilePreview) => {
+    if (isAudio(fp)) {
+      return <Preview onClick={() => handlePrevClick(fp)} fp={fp} icon="volume-2" />;
+    }
+
+    if (isVideo(fp)) {
+      return <Preview onClick={() => handlePrevClick(fp)} fp={fp} icon="youtube" />;
```
Comment (kespinola) [inline src/common/components/elements/NFTPreviewGrid.tsx:175]: There isn't a general video play icon? Don't want to use youtube as our play button.

File: src/modules/nfts/components/wizard/Upload.tsx [hunk 5]
```diff
@@ -81,7 +90,19 @@ export default function UploadStep({ nextStep, dispatch, images }: Props) {
...
-      const { status } = info.file;
+      const { status, size, name } = info.file;
+
+      console.log('file is ', info.file);
+      if (size && size > MAX_FILE_SIZE) {
+        window.alert(
+          `The file name ${name} you are trying to upload is ${(size / 1000000).toFixed(
+            0
+          )}MB, only files equal to or under ${MAX_FILE_SIZE / 1000000}MB are allowed`
+        );
```
Comment (kespinola) [inline src/modules/nfts/components/wizard/Upload.tsx:101]: Update later to antd notification instead of window alert.

**Review by kespinola — APPROVED** [review]
Some recommendations but I think the most meaningful follow up would be separating cover image upload from the value transform.

File: src/common/components/elements/NFTPreviewGrid.tsx (hunk 4 above)
Comment (davetorbeck) [reply src/common/components/elements/NFTPreviewGrid.tsx:175]: @damianr Requested `youtube over `video`.  What's the concern?

**Pull request merged**